### PR TITLE
Add validation for input properties, refactor, add unit tests

### LINKED ISSUE #135: Validate properties
The input properties should be validated:
- `service-dns-timeout`: cannot be negative
- `kubernetes-api-retries`: cannot be negative
- If DNS Lookup parameters are specified, the Kubernetes API cannot be specified and vice versa
- ...

It may also be possible to refactor parameters validation and parsing to a separate class.

### PR DESCRIPTION
**Pull request by leszko**
Changes:
- Refactor handling input properties to a separate class `KubernetesConfig`
- Add validation for the input properties (i.e. non-negative values, enabled either Kubernetes API or DNS Lookup)
- Add Unit tests

fix #135

### CODE REVIEW

File: src/main/java/com/hazelcast/kubernetes/KubernetesConfig.java [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+                + "service-port: " + servicePort + ", "
+                + "service-label: " + serviceLabelName + ", "
+                + "service-label-value: " + serviceLabelValue + ", "
+                + "namespace: " + namespace + ", "
+                + "resolve-not-ready-addresses: " + resolveNotReadyAddresses + ", "
+                + "kubernetes-api-retries: " + kubernetesApiRetries + ", "
+                + "kubernetes-master: " + kubernetesMasterUrl + "}";
+    }
+
+    enum Mode {
```
Comment (mesutcelik) [inline src/main/java/com/hazelcast/kubernetes/KubernetesConfig.java]: DiscoveryMode can be more descriptive here

File: src/main/java/com/hazelcast/kubernetes/KubernetesConfig.java [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+        this.kubernetesCaCertificate = caCertificate(properties);
+        this.servicePort = getOrDefault(properties, KUBERNETES_SYSTEM_PREFIX, SERVICE_PORT, 0);
+
+        validateConfig();
+    }
+
+    private String getNamespaceOrDefault() {
+        String namespace = System.getenv("KUBERNETES_NAMESPACE");
+        if (namespace == null) {
+            namespace = System.getenv("OPENSHIFT_BUILD_NAMESPACE");
```
Comment (mesutcelik) [inline src/main/java/com/hazelcast/kubernetes/KubernetesConfig.java]: Why do we have openshift specific customization here?

Comment (leszko) [reply src/main/java/com/hazelcast/kubernetes/KubernetesConfig.java]: I actually never used these variables, but looking at [this](https://docs.openshift.com/container-platform/3.4/dev_guide/builds/build_output.html) my guess is that the OpenShift variable is automatically set when you create an image stream with OpenShift tooling.

File: src/main/java/com/hazelcast/kubernetes/KubernetesConfig.java (hunk 1 above)
Comment (leszko) [reply src/main/java/com/hazelcast/kubernetes/KubernetesConfig.java]: Changed. Thanks!

**Pull request merged**